src/main/java/bio/terra/service/job/JobBuilder.java

```diff
@@ -23,35 +25,41 @@ public JobBuilder(
 
     // initialize with required parameters
     this.jobParameterMap = new FlightMap();
-    jobParameterMap.put(JobMapKeys.DESCRIPTION.getKeyName(), description);
-    jobParameterMap.put(JobMapKeys.REQUEST.getKeyName(), request);
-    jobParameterMap.put(JobMapKeys.AUTH_USER_INFO.getKeyName(), userReq);
-    jobParameterMap.put(JobMapKeys.SUBJECT_ID.getKeyName(), userReq.getSubjectId());
+    JobMapKeys.DESCRIPTION.put(jobParameterMap, description);
+    JobMapKeys.REQUEST.put(jobParameterMap, request);
+    JobMapKeys.AUTH_USER_INFO.put(jobParameterMap, userReq);
+    JobMapKeys.SUBJECT_ID.put(jobParameterMap, userReq.getSubjectId());
   }
 
+  private static final Set<JobMapKeys> REQUIRED_KEYS =
+      EnumSet.of(
+          JobMapKeys.DESCRIPTION,
+          JobMapKeys.REQUEST,
+          JobMapKeys.AUTH_USER_INFO,
+          JobMapKeys.SUBJECT_ID);
+
   // use addParameter method for optional parameter
   // returns the JobBuilder object to allow method chaining
-  public JobBuilder addParameter(String keyName, Object val) {
-    if (keyName == null) {
-      throw new InvalidJobParameterException("Parameter name cannot be null.");
-    }
-
+  public JobBuilder addParameter(JobMapKeys key, Object val) {
     // check that keyName doesn't match one of the required parameter names
     // i.e. disallow overwriting one of the required parameters
-    if (keyName.equals(JobMapKeys.DESCRIPTION.getKeyName())
-        || keyName.equals(JobMapKeys.REQUEST.getKeyName())
-        || keyName.equals(JobMapKeys.AUTH_USER_INFO.getKeyName())
-        || keyName.equals((JobMapKeys.SUBJECT_ID.getKeyName()))) {
+    if (REQUIRED_KEYS.contains(key)) {
       throw new InvalidJobParameterException(
-          "Required parameters can only be set by the constructor. (" + keyName + ")");
+          "Required parameters can only be set by the constructor. (" + key + ")");
     }
 
     // note that this call overwrites a parameter if it already exists
-    jobParameterMap.put(keyName, val);
+    key.put(jobParameterMap, val);
 
     return this;
   }
 
+  public JobBuilder addParameter(String key, Object val) {
+    // note that this call overwrites a parameter if it already exists
+    jobParameterMap.put(key, val);
```

Comment:
What happens if key is `null`? Before there was a null check.

Reply:
This is placeholder code to get things to compile until all keys are replaced with objects. Even so, I wonder how the `null` error could ever be triggered; the arguments are fixed and aren't based on inputs or any other variables. The underlying code will eventually NPE when it calls `Map.put()` with a null key, so I'm not sure what benefit there is for having an explicit check here.